From: Osamu Aoki <osamu@debian.org>
Date: Sat, 4 May 2019 15:56:26 +0900
Subject: Use UTF-8 for zipcode conversion (essential change) (code)

* src-worddic/ext_ent.c (pushback_place_name): Use UTF-8.

This is taken from a commit in the upstream's development branch
containing no auto-generated file (branch=master) of salsa(NIIBE Yutaka).
Somehow, this was not included in the branch=release where the upstream
released tarball contents was intended to be tracked (but ... that
doesn't seem to happen).

Original commit:
97f5ef0 ("Use UTF-8 for zipcode conversion.", 2015-04-28)

Space adjusted and dropped non-essential changes to ZIPDICT_EUC.
(I skipped all attempts to change ZIPDICT_EUC to ZIPDICT..
 That should be done after discussion with previous maintainers.
 This is intended to be a pure bug fix patch.)

The change of encoding setting in ext_ent.c should fix problem with
7 digits numbers returning odd results.
      * 1000000 -> 〓延根〓遵〓篁ｇ〓〓〓
      * 2270000 -> 腑〓〓綏〓〓罔〓〓絽〓〓〓〓〓
      * 2270001 -> No problem
      * ...
      * 2270030 -> No problem
      * 2270031 -> 腑〓〓綏〓〓罔〓〓絽〓〓〓〓〓絲阪〓〓〓
      * 2270032 -> 腑〓〓綏〓〓罔〓〓絽〓〓〓〓〓〓〓〓〓〓
      * ...
      * 2270044 -> 腑〓〓綏〓〓罔〓〓絽〓〓〓〓〓〓〓〓〓〓

Signed-off-by: Osamu Aoki <osamu@debian.org>
---
 src-worddic/ext_ent.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src-worddic/ext_ent.c b/src-worddic/ext_ent.c
index 1ebf910..5a7ab7e 100644
--- a/src-worddic/ext_ent.c
+++ b/src-worddic/ext_ent.c
@@ -56,7 +56,7 @@ pushback_place_name(struct zipcode_line *zl, char *pn)
     return ;
   }
   zl->strs = realloc(zl->strs, sizeof(xstr *) * (zl->nr + 1));
-  zl->strs[zl->nr] = anthy_cstr_to_xstr(pn, ANTHY_EUC_JP_ENCODING);
+  zl->strs[zl->nr] = anthy_cstr_to_xstr(pn, ANTHY_UTF8_ENCODING);
   zl->nr++;
 }
 
